client/dashboard/app/queries/domain-nameservers.ts

```diff
@@ -0,0 +1,18 @@
+import { queryOptions, mutationOptions } from '@tanstack/react-query';
+import { fetchDomainNameservers, updateDomainNameservers } from '../../data/domain-nameservers';
+import { queryClient } from '../query-client';
+
+export const domainNameserversQuery = ( domainName: string ) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'nameservers' ],
+		queryFn: () => fetchDomainNameservers( domainName ),
+		refetchOnWindowFocus: false,
+	} );
+
+export const domainNameserversUpdateMutation = ( domainName: string ) =>
```

Comment:
[nit] we don't usually put `Update` in the mutation name for simplicity -- it's already implied. 😄 


```suggestion
export const domainNameserversMutation = ( domainName: string ) =>
```